src/toil/lib/aws/__init__.py

```diff
@@ -25,3 +25,12 @@ def client(*args, **kwargs):
 def resource(*args, **kwargs):
     import boto3
     return boto3.resource(*args, **kwargs)
+
+
+def zone_to_region(zone: str):
+    """Get a region (e.g. us-west-2) from a zone (e.g. us-west-1c)."""
+    availability_zone = re.compile(r'^([a-z]{2}-[a-z]+-[1-9][0-9]*)([a-z])$')
```

Comment:
I think we had this outside the function so we wouldn't recompile it all the time. But maybe `re` memoizes these internally.

Reply:
I believe it is: https://stackoverflow.com/questions/12514157/how-does-pythons-regex-pattern-caching-work

Reply:
I've added a comment.